File: RR.Frontend/src/DTOs/Address.ts
export class Address {
  country: string;
  region: string;
  postalCode: string;
  city: string;
  streetAddress: string;
  note: string;
  
  constructor() {
    this.country = '';
    this.region = '';
    this.postalCode = '';
    this.city = '';
    this.streetAddress = '';
    this.note = '';
  }

  static from(obj: any): Address {
    return Object.assign(new Address(), obj);
  }
  
  isEmpty(): boolean {
    return (!this.country || this.country.trim() === '') &&
      (!this.region || this.region.trim() === '') &&
      (!this.postalCode || this.postalCode.trim() === '') &&
      (!this.city || this.city.trim() === '') &&
      (!this.streetAddress || this.streetAddress.trim() === '') &&
      (!this.note || this.note.trim() === '');
  }
}

File: RR.Common/Models/Address.cs
namespace RR.Common.Models;

public class Address : IOwnable
{
    public int? Id { get; set; }
    public int? GroupId { get; set; }
    public string? Country { get; set; }
    public string? Region { get; set; }
    public string? PostalCode { get; set; }
    public string City { get; set; }
    public string StreetAddress { get; set; }
    public string? Note { get; set; }
}
